src/main/kotlin/no/nav/familie/ba/sak/integrasjoner/økonomi/utbetalingsoppdrag/BehandlingsinformasjonUtleder.kt

```diff
@@ -0,0 +1,86 @@
+package no.nav.familie.ba.sak.integrasjoner.økonomi.utbetalingsoppdrag
+
+import no.nav.familie.ba.sak.common.ClockProvider
+import no.nav.familie.ba.sak.kjerne.beregning.domene.AndelTilkjentYtelse
+import no.nav.familie.ba.sak.kjerne.beregning.domene.TilkjentYtelse
+import no.nav.familie.ba.sak.kjerne.fagsak.Fagsak
+import no.nav.familie.ba.sak.kjerne.fagsak.FagsakType
+import no.nav.familie.ba.sak.kjerne.vedtak.Vedtak
+import no.nav.familie.felles.utbetalingsgenerator.domain.Behandlingsinformasjon
+import no.nav.familie.felles.utbetalingsgenerator.domain.IdentOgType
+import org.springframework.stereotype.Component
+import java.time.LocalDate
+import java.time.YearMonth
+
+@Component
+class BehandlingsinformasjonUtleder(
+    private val endretMigreringsdatoUtleder: EndretMigreringsdatoUtleder,
+    private val clockProvider: ClockProvider,
+) {
+    fun utled(
+        saksbehandlerId: String,
+        vedtak: Vedtak,
+        forrigeTilkjentYtelse: TilkjentYtelse?,
+        sisteAndelPerKjede: Map<IdentOgType, AndelTilkjentYtelse>,
+        erSimulering: Boolean,
+    ): Behandlingsinformasjon {
+        val behandling = vedtak.behandling
+        val fagsak = behandling.fagsak
+        val endretMigreringsdato = endretMigreringsdatoUtleder.utled(fagsak, forrigeTilkjentYtelse)
+        return Behandlingsinformasjon(
+            saksbehandlerId = saksbehandlerId,
+            behandlingId = behandling.id.toString(),
+            eksternBehandlingId = behandling.id,
+            eksternFagsakId = fagsak.id,
+            fagsystem = FagsystemBA.BARNETRYGD,
+            personIdent = fagsak.aktør.aktivFødselsnummer(),
+            vedtaksdato = vedtak.vedtaksdato?.toLocalDate() ?: LocalDate.now(clockProvider.get()),
+            opphørAlleKjederFra = finnOpphørsdatoForAlleKjeder(forrigeTilkjentYtelse, sisteAndelPerKjede, endretMigreringsdato),
+            utbetalesTil = finnUtebetalesTil(fagsak),
+            opphørKjederFraFørsteUtbetaling = finnOpphørKjederFraFørsteUtbetaling(endretMigreringsdato, erSimulering),
+        )
+    }
+
+    private fun finnOpphørsdatoForAlleKjeder(
+        forrigeTilkjentYtelse: TilkjentYtelse?,
+        sisteAndelPerKjede: Map<IdentOgType, AndelTilkjentYtelse>,
+        endretMigreringsdato: YearMonth?,
+    ): YearMonth? {
+        if (forrigeTilkjentYtelse == null || sisteAndelPerKjede.isEmpty()) {
+            return null
+        }
+        if (endretMigreringsdato != null) {
+            return endretMigreringsdato
+        }
+        return null
+    }
```

Comment:
```suggestion
    private fun finnOpphørsdatoForAlleKjeder(
        forrigeTilkjentYtelse: TilkjentYtelse?,
        sisteAndelPerKjede: Map<IdentOgType, AndelTilkjentYtelse>,
        endretMigreringsdato: YearMonth?,
    ): YearMonth? =
        when {
        forrigeTilkjentYtelse == null || sisteAndelPerKjede.isEmpty() -> null
        endretMigreringsdato != null -> endretMigreringsdato
        else -> null
    }
```

Foretrekker when når man skal sjekke på flere kondisjoner og returnere forskjellig basert på dette. 
Smak og behag, men jeg synes at når vi har muligheten til å bruke when og bruke verdien av det direkte istedenfor flere 'return' så er det foretrekkbart

Reply:
Kan ikke dette skrives som under? 

```
        if (forrigeTilkjentYtelse == null || sisteAndelPerKjede.isEmpty()) {
            return null
        }
        return endretMigreringsdato
```

Reply:
Ja, enig! ⭐ 
Det er nok foretrekkbart vs som det er i dag